src/collective/cover/layout.py

```diff
@@ -233,6 +242,112 @@ def update(self):
                 i += 1
 
 
+# Ok we exclude a little too much here. We exclude "-89", which is invalid, but
+# we also exclude "-abc" which IS a valid class, but it's also a silly CSS
+# class, so why bother to support it!?
+VALID_CSS_RE = re.compile('[a-zA-Z_][0-9a-zA-Z_-]*')
+
+def cssConstraint(value):
+    """Check for valid CSS names in a space separated string
+    """
+    invalidTokens = []
+    for token in value.split():
+        if not VALID_CSS_RE.match(token):
+            invalidTokens.append(token)
+
+    if invalidTokens:
+        raise Invalid(_(u"The following strings are not valid css identifiers: %s" %
+                        ' '.join(invalidTokens)))
+    return True
+
+
+class IRowConfigureForm(form.Schema):
+    """ Popout form to update row configuration (currently just classes)"""
+
+    form.mode(rownum='hidden')
+    rownum = schema.Int(
+            default=-1,
+        )
+
+    row_classes = schema.TextLine(
+            title=_(u"Row Classes"),
+            description=_(u"CSS class(es) for Cover row - space separated"),
+            constraint=cssConstraint,
+        )
+
+
+class RowConfigureForm(form.SchemaForm):
+    """ Configure settings for Cover Row
+
+    """
+    grok.context(ICover)
+    grok.name('configure-row')
+    grok.require('cmf.ModifyPortalContent')
+
+    schema = IRowConfigureForm
+    ignoreContext = True
+    name = 'configure_row'
+
+
+    def updateWidgets(self):
```

Comment:
Follow PEP8 for the name of methods https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables